crates/matrix-sdk-crypto/src/identities/user.rs

```diff
@@ -376,11 +410,93 @@ impl UserIdentityData {
 /// This is the user identity of a user that isn't our own. Other users will
 /// only contain a master key and a self signing key, meaning that only device
 /// signatures can be checked with this identity.
+///
+/// This struct also contains the currently pinned master key for that user.
+/// The first time a cryptographic identity is seen for a given user, it
+/// will be associated to that user (i.e. pinned). Future interactions
+/// will expect this user crypto identity to stay the same,
+/// this will help prevent some MITM attacks.
+/// In case of identity change, it will be possible to pin the new identity
+/// is the user wants.
```

Comment:
```suggestion
/// This struct also contains the currently pinned user identity (master signing
/// key) for that user.
///
/// The first time a cryptographic user identity is seen for a given user, it
/// will be associated with that user ("pinned"). Future interactions
/// will expect this identity to stay the same, to avoid MITM attacks from the
/// homeserver.
///
/// The user can explicitly pin the new identity to allow for legitimate
/// identity changes (for example, in case of key material or device loss).
```